lib/APPReinforcementLearning/src/DrivingState.h

```diff
@@ -91,114 +90,34 @@ class DrivingState : public IState
      */
     void exit() final;
 
-protected:
-private:
     /**
-     * The line detection status.
+     * Set target motor speeds.
+     *
+     * @param[in] leftMotor  Left motor speed. [steps/s]
+     * @param[in] rightMotor Right motor speed. [steps/s]
      */
-    enum LineStatus
-    {
-        LINE_STATUS_NO_START_LINE_DETECTED = 0, /**< No start line detected. */
-        LINE_STATUS_START_LINE_DETECTED,        /**< Start line detected. */
-        LINE_STATUS_STOP_LINE_DETECTED          /**< Stop line detected. */
-    };
+    void setTargetSpeeds(int16_t leftMotor, int16_t rightMotor);
 
     /**
-     * The track status.
+     * Check the abort conditions while driving the challenge.
+     *
+     * @return If abort is required, it will return true otherwise false.
      */
-    enum TrackStatus
-    {
-        TRACK_STATUS_NORMAL = 0,              /**< Normal line conditions. */
-        TRACK_STATUS_START_STOP_LINE,         /**< Driving over start-/stop-line. */
-        TRACK_STATUS_RIGHT_ANGLE_CURVE_LEFT,  /**< Right angle curve to the left detected. */
-        TRACK_STATUS_RIGHT_ANGLE_CURVE_RIGHT, /**< Right angle curve to the right detected. */
-        TRACK_STATUS_SHARP_CURVE_LEFT,        /**< Sharp curve to the left detected. */
-        TRACK_STATUS_SHARP_CURVE_RIGHT,       /**< Sharp curve to the right detected. */
-        TRACK_STATUS_SHARP_CURVE_LEFT_TURN,   /**< Sharp curve to the left turns now. */
-        TRACK_STATUS_SHARP_CURVE_RIGHT_TURN,  /**< Sharp curve to the right turns now. */
-        TRACK_STATUS_TRACK_LOST_BY_GAP,       /**< Track lost by gap. */
-        TRACK_STATUS_TRACK_LOST_BY_MANOEUVRE, /**< Track lost by driving manoeuvre. */
-        TRACK_STATUS_FINISHED                 /**< Robot found the end line or a error happened. */
-    };
+    bool isAbortRequired();
 
+protected:
+private:
     /** Observation duration in ms. This is the max. time within the robot must be finished its drive. */
     static const uint32_t OBSERVATION_DURATION = 3000000;
```

Comment:
How do you get this number? Can you decompose it in easier numbers to work with? (Minutes, for example)

Reply:
I have  used the same number as for Linefollower Application !

Reply:
Ok